examples/classification/benchmark.py

```diff
@@ -0,0 +1,393 @@
+        return (
+            model_path,
+            None,
+            original_image_shape,
+        )  # model graph does not have static integer input shape
+
+    if image_shape is None or original_image_shape == tuple(image_shape):
+        return model_path, None, original_image_shape  # no shape modification needed
+
+    set_tensor_dim_shape(model_input, 2, image_shape[0])
+    set_tensor_dim_shape(model_input, 3, image_shape[1])
+
+    tmp_file = NamedTemporaryFile()  # file will be deleted after program exit
+    onnx.save(model, tmp_file.name)
+
+    print(
+        f"Overwriting original model shape {original_image_shape} to {image_shape}\n"
+        f"Original model path: {original_model_path}, new temporary model saved to "
+        f"{tmp_file.name}"
+    )
+
+    return tmp_file.name, tmp_file, image_shape
+
+
+def _load_model(
+    model_filepath: str,
+    batch_size: int,
+    num_cores: Optional[int],
+    num_sockets: Optional[int],
+    engine: str,
+    image_shape: Tuple[int, int],
+):
+    # load and return respective classification model according to arguments
+
+    if (
+        num_cores is not None
+        and engine == ORT_ENGINE
+        and onnxruntime.__version__ < "1.7"
+    ):
+        raise ValueError(
+            "overriding default num_cores not supported for onnxruntime < 1.7.0. "
+            "If using an older build with OpenMP, try setting the OMP_NUM_THREADS "
+            "environment variable"
+        )
+    if num_sockets is not None and engine != DEEPSPARSE_ENGINE:
+        raise ValueError(f"Overriding num_sockets is not supported for {engine}")
+
+    # scale static ONNX graph to desired image shape
+    if engine in [DEEPSPARSE_ENGINE, ORT_ENGINE]:
+        model_filepath, _, image_shape = fix_onnx_input_shape(
+            model_filepath, image_shape
+        )
+
+    # load model
+    if engine == DEEPSPARSE_ENGINE:
+        print(f"Compiling deepsparse model for {model_filepath}")
+        model = compile_model(model_filepath, batch_size, num_cores, num_sockets)
+        print(f"Engine info: {model}")
+
+    elif engine == ORT_ENGINE:
+        print(f"loading onnxruntime model for {model_filepath}")
+
+        sess_options = onnxruntime.SessionOptions()
+        if num_cores is not None:
+            sess_options.intra_op_num_threads = num_cores
+        sess_options.log_severity_level = 3
+        sess_options.graph_optimization_level = (
+            onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL
+        )
+
+        onnx_model = onnx.load(model_filepath)
+        override_model_batch_size(onnx_model, batch_size)
+        model = onnxruntime.InferenceSession(
+            onnx_model.SerializeToString(), sess_options=sess_options
+        )
+
+    return model, image_shape
+
+
+def _parse_device(device: Union[str, int]) -> Union[str, int]:
+    try:
+        return int(device)
+    except Exception:
+        return device
+
+
+def _add_and_parse_args(arguments=None):
```

Comment:
nit: combine with `_get_parser()` to a single parse args function

Reply:
👍